### DISCUSSION TITLE
Pet Shop error: Contract has not been deployed to detected network (network/artifact mismatch)

### DISCUSSION THREAD

**Discussion by DaikyXendo** [ConsenSys-archive/truffle#5358] [General Q&A]
I followed your *Pete's Pet Shop* tutorial however when I run ```npm run dev``` I get this error on the console window. Where did I go wrong? I double checked the code files and everything is fine.
![image](https://user-images.githubusercontent.com/46809008/180902726-6545abfd-6865-4755-b783-ae88813257b8.png)

**Comment by leeftk** [comment]
Hey @DaikyXendo thanks for opening this discussion. You may want to take another look at the section that says migration and verify that you followed all the steps for migrating you contracts to a local blockchain (`ganache`). If you already have a migrate file I'd try running `truffle migrate` in your root directory to migrate your contracts and then see if this worked. Let me know if that helps!
